src/main/java/com/joss/conductor/mobile/annotation/PlatformFindByHelper.java

```diff
@@ -11,41 +12,45 @@
-                            throw new ClassCastException(field.getName() + " must be of type By!");
+                        if (annotation instanceof AndroidFindBy || annotation instanceof IOSFindBy) {
+                            if (!field.getType().equals(By.class)) {
+                                throw new PlatformFindByException(field.getName() + " must be of type By!");
+                            }
                         }
-                    }
 
-                    if (annotation instanceof AndroidFindBy && platform.equals(Platform.ANDROID)) {
-                        AndroidFindBy androidFindByAnnotation = (AndroidFindBy) annotation;
-                        if (!androidFindByAnnotation.value().isEmpty()) {
-                            field.set(page, By.id(androidFindByAnnotation.value()));
-                        } else if (!androidFindByAnnotation.xpath().isEmpty()) {
-                            field.set(page, By.xpath(androidFindByAnnotation.xpath()));
+                        if (annotation instanceof AndroidFindBy && platform.equals(Platform.ANDROID)) {
+                            AndroidFindBy androidFindByAnnotation = (AndroidFindBy) annotation;
+                            if (!androidFindByAnnotation.value().isEmpty()) {
+                                field.set(page, By.id(androidFindByAnnotation.value()));
+                            } else if (!androidFindByAnnotation.xpath().isEmpty()) {
+                                field.set(page, By.xpath(androidFindByAnnotation.xpath()));
+                            }
                         }
-                    }
 
-                    if (annotation instanceof IOSFindBy && platform.equals(Platform.IOS)) {
-                        IOSFindBy iOSFindByAnnotation = (IOSFindBy) annotation;
-                        if (!iOSFindByAnnotation.value().isEmpty()) {
-                            field.set(page, By.name(iOSFindByAnnotation.value()));
-                        } else if (!iOSFindByAnnotation.xpath().isEmpty()) {
-                            field.set(page, By.xpath(iOSFindByAnnotation.xpath()));
+                        if (annotation instanceof IOSFindBy && platform.equals(Platform.IOS)) {
+                            IOSFindBy iOSFindByAnnotation = (IOSFindBy) annotation;
+                            if (!iOSFindByAnnotation.value().isEmpty()) {
+                                field.set(page, By.name(iOSFindByAnnotation.value()));
+                            } else if (!iOSFindByAnnotation.xpath().isEmpty()) {
+                                field.set(page, By.xpath(iOSFindByAnnotation.xpath()));
+                            }
                         }
                     }
                 }
-            }
-            current = current.getSuperclass();
-        } while (current != null);
+                current = current.getSuperclass();
+            } while (current != null);
+        } catch (Exception e) {
```

Comment:
Do we need this outer try/catch?

Reply:
yes, `field.set()` throws IllegalAccessException.